### PR TITLE
adds reviews to project show page and tests project/review associations

### PR DESCRIPTION
**Pull request by NicoleCarpenter**

### CODE REVIEW

**Comment by hnlee** [comment]
👍

File: spec/integration/review_spec.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+require 'spec_helper'
+
+describe 'project show page with reviews' do
+  it 'shows all reviews in a project' do
+    project = create(:project, title: "Java Tic-Tac-Toe")
+    review = create(:review, content: "Looks good!", project_id: project.id)
+
+    visit "/reviews"
```
Comment (JPhoenx) [inline spec/integration/review_spec.rb]: I think that reviews should live on the project show page like so:

visit '/projects/' + project.id.to_s

File: spec/controllers/reviews_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+require 'spec_helper'
+
+RSpec.describe ReviewsController, :type => :controller do
```
Comment (JPhoenx) [inline spec/controllers/reviews_controller_spec.rb]: I think this should be a nested resource:

I can't remember how to do it off the top of my head though.

Something like ProjectsController::ReviewsController

or something.

File: config/routes.rb [hunk 3]
```diff
@@ -1,4 +1,5 @@
 Rails.application.routes.draw do
   root 'projects#index'
   resources :projects, only: [:index]
+  resources :reviews, only: [:index]
```
Comment (JPhoenx) [inline config/routes.rb]: No route for this yet.

File: app/views/reviews/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+<% @reviews.each do |review| %>
```
Comment (JPhoenx) [inline app/views/reviews/index.html.erb]: Throw this logic onto projects show page.

File: app/models/review.rb [hunk 5]
```diff
@@ -0,0 +1,6 @@
+class Review < ApplicationRecord
+  validates :content, presence: true
+
+  belongs_to :project
```
Comment (JPhoenx) [inline app/models/review.rb]: We need a test for these

File: app/controllers/reviews_controller.rb [hunk 6]
```diff
@@ -0,0 +1,5 @@
+class ReviewsController < ApplicationController
+  def index
```
Comment (JPhoenx) [inline app/controllers/reviews_controller.rb]: Remove

File: spec/models/review_spec.rb [hunk 7]
```diff
@@ -0,0 +1,11 @@
+require 'rails_helper'
+
+RSpec.describe Review do
+  it 'has content' do
+    review = Review.new(content: "Looks good!",
+                        project_id: 1)
```
Comment (NicoleCarpenter) [inline spec/models/review_spec.rb]: 👍

File: spec/integration/review_spec.rb (hunk 1 above)
Comment (NicoleCarpenter) [inline spec/integration/review_spec.rb]: 👍

File: spec/controllers/reviews_controller_spec.rb (hunk 2 above)
Comment (NicoleCarpenter) [inline spec/controllers/reviews_controller_spec.rb]: 👍

File: db/seeds.rb [hunk 8]
```diff
@@ -33,4 +33,31 @@
...
+            'This work is really good, I can tell that you have made a lot of changes from the previous iteration and it shows. You addressed the spacing issue, but I am not sure that my feedback about the font was taken correctly. Maybe we can do an in person meeting and I can explain what I meant.', 
+            'Wow!', 
+            'You should move line 43 to a separate file, it is creating a clear violation of the Single Responsibility Principle', 
+            'By including line 112, you are creating a violation of the Dependency Inversion Principle', 
+            'I am really impressed by your use of the Open-Closed Principle',
+            'Great use of the Liskov Substitution Principle here', 
+            'The way that this is written, it violates the Interface Segregation Principle. I would fix this if I were you.',
+            'This looks really great. It is obvious that you are really considering your SOLID principles here']
+
+35.times do
```
Comment (NicoleCarpenter) [inline db/seeds.rb]: I was trying to have more reviews than projects, so I am going to leave this as is

File: db/schema.rb [hunk 9]
```diff
@@ -22,4 +22,13 @@
     t.datetime "updated_at",  null: false
   end
 
+  create_table "reviews", force: :cascade do |t|
+    t.string   "content"
+    t.integer  "project_id"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.index ["project_id"], name: "index_reviews_on_project_id", using: :btree
```
Comment (NicoleCarpenter) [inline db/schema.rb]: Not sure either, this was automatic, I think it is harmless

File: config/routes.rb (hunk 3 above)
Comment (NicoleCarpenter) [inline config/routes.rb]: 👍  deleted

File: app/views/reviews/index.html.erb (hunk 4 above)
Comment (NicoleCarpenter) [inline app/views/reviews/index.html.erb]: 👍

File: app/controllers/reviews_controller.rb (hunk 6 above)
Comment (NicoleCarpenter) [inline app/controllers/reviews_controller.rb]: 👍

File: app/models/review.rb (hunk 5 above)
Comment (NicoleCarpenter) [inline app/models/review.rb]: 👍

**Comment by JPhoenx** [comment]
I wish I had more thumbs so I could thumbs up this moar.

👍 👍

File: db/schema.rb (hunk 9 above)
Comment (JPhoenx) [inline db/schema.rb]: 👍

File: db/seeds.rb (hunk 8 above)
Comment (JPhoenx) [inline db/seeds.rb]: 👍

**Comment by hnlee** [comment]
👍

**Pull request merged**